core/src/main/java/cz/xtf/core/git/GitRemoteResolver.java

```diff
@@ -0,0 +1,140 @@
+package cz.xtf.core.git;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.eclipse.jgit.transport.RemoteConfig;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Try to resolve repository remote URL and branch from .git directory
+ * <p>
+ * This method tries to match HEAD commit to remote references.
+ * If there is a match, the remote URL and branch are set.
+ * For attached HEAD this method could be simplified with jgit methods, but
+ * "universal" approach was chosen, as for example Jenkins git plugin creates a detached state.
+ * In case of multiple matches, upstream or origin (in this order) is preferred.
+ * </p>
+ */
+@Slf4j
+class GitRemoteResolver {
+    private static final String URL_TEMPLATE = "https://%s/%s/%s";
+    private static String reference;
+    private static String url;
+
+    /**
+     * We require HTTPS format, for unauthorized access to the repository, let's convert it
+     */
+    private static String getRepositoryUrl(String host, String remote, String repository) {
+        return String.format(URL_TEMPLATE, host, remote, repository);
+    }
+
+    /**
+     * Try to set repository ref and URL from HEAD commit
+     */
+    private static void resolveRepoFromHEAD() throws IOException, URISyntaxException {
+        //look for a git repository recursively till system root folder
+        Repository repository = new FileRepositoryBuilder().findGitDir().build();
```

Comment:
Would it make sense to use `findGitDir(File current)` instead and make `current` configurable?

Reply:
IMO not really, at this point user could specify `xtf.git.repository.url` which is also more universal than "hardcoding" local path